src/frontend/src/handler/create_source.rs

```diff
@@ -87,6 +89,10 @@ pub async fn handle_create_source(
 ) -> Result<PgResponse> {
     let with_properties = handle_with_properties("create_source", stmt.with_properties.0)?;
 
+    let (column_descs, pk_column_id_from_columns) = bind_sql_columns(stmt.columns)?;
+    let (columns, pk_column_ids) =
+        bind_sql_table_constraints(&column_descs, pk_column_id_from_columns, stmt.constraints)?;
```

Comment:
How about leaving a TODO here about the `_row_id` column?

Reply:
Added to `bind_sql_table_constraints`, which is the source of truth for `pk_column_ids` and in charge of adding/removing the extra row id column.